File: Linear search/BinarySearch.cpp
#include<iostream>
using namespace std;

int binarysearch(int a[],int num,int start,int last)
{
    int mid;
    if(start<=last)
    {
        mid=start+((last-start)/2);
        if(a[mid]==num)
            return mid;
        else if(num<a[mid])
            return binarysearch(a,num,start,mid-1);
        else if(num>a[mid])
            return binarysearch(a,num,mid+1,last);
    }
    return -1;
}
int main()
{
    int siz,num,pos;
    cout<<"Enter the size of list :";
    cin>>siz;
    int *a=new int[siz];

    cout<<"\nEnter elements :";
    for(int i=0;i<siz;i++)
        cin>>a[i];

    cout<<"\n Enter the element to be searched:";
    cin>>num;

    pos=binarysearch(a,num,0,siz-1);
    if(pos==-1)
        cout<<"\n Element not found";
    else
        cout<<"\n Element found at position :"<<pos+1;
    return 0;
}

File: Searching/BinarySearch.java
class BinarySearch {
  public static int binarySearch(int a[], int x) {
    int n = a.length;
    int l=0,r=n-1;
    while(l<=r){
        int mid = l+(r-l)/2;
        if(a[mid]==x)return mid;
        else if(a[mid]>x)r=mid-1;
        else l=mid+1;
    }
    return -1;
  }

  public static void main(String args[]) {
  int array[] = { 10, 20, 30, 40, 50, 60 };
  int x = 50;

  int result = binarySearch(array, x);

  if (result == -1)
    System.out.print("Element not found");
  else
    System.out.print("Element found at index: " + result);
  }
}